server/src/main/java/org/opensearch/cluster/metadata/RepositoryMetadata.java

```diff
@@ -50,6 +53,7 @@
 @PublicApi(since = "1.0.0")
 public class RepositoryMetadata implements Writeable {
 
+    private final static Set<String> IGNORED_SETTINGS = Set.of(PREFIX_MODE_VERIFICATION_SETTING.getKey());
```

Comment:
why are we ignoring this ? If we do this then `registerOrUpdateRepository` won't consider this as an update .

Reply:
> why are we ignoring this ? 

`ensureRemoteStoreNodesCompatibility`checks if the repository details of the joining node is same or not when compared with existing node/existing repository. The current logic, as part of the compatibility check, verifies that the `settings` of the new repository and existing repository are the same. The new setting plays role only for the prefix mode verification of repository during bootstrap and plays no other roles elsewhere. Without this change, the new node that is joining an existing cluster fails to join it.
https://github.com/opensearch-project/OpenSearch/blob/4abcf395897fda619a186afe53fbb5cfaa34b841/server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java#L574-L583

> If we do this then registerOrUpdateRepository won't consider this as an update .

That's right probably. IMO this is fine to do so.

Reply:
But this code won't be deployed on the older nodes . So it would anyways not allow the node having this setting on .

Reply:
This does makes sense. Let me see if there is any other way around.